### PR TITLE
Refactor insertBufferNode and add unit tests

### PR DESCRIPTION
**Pull request by congqixia**
Refactor long function in `insertBufferNode`
Add some unit tests

See also #6357

Signed-off-by: Congqi Xia <congqi.xia@zilliz.com>

### CODE REVIEW

File: internal/datanode/flow_graph_insert_buffer_node_test.go [hunk 1]
```diff
@@ -560,3 +562,137 @@ func TestFlowGraphInsertBufferNode_AutoFlush(t *testing.T) {
 
 	})
 }
+
+// CompactedRootCoord has meta info compacted at ts
+type CompactedRootCoord struct {
```
Comment (XuanYang-cn) [inline internal/datanode/flow_graph_insert_buffer_node_test.go:567]: dup to RootCoordFactory  in  mock_test.go

Comment (congqixia) [reply internal/datanode/flow_graph_insert_buffer_node_test.go:567]: CompactedRootCoord composes with `RootCoordFactory` when used and only override `DescribeCollection` behavior

**Pull request merged**